L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -0,0 +1,327 @@
+
+        /// Build a temporary Stub
+        TTStub< Ref_Phase2TrackerDigi_ > tempTTStub( stackDetid );
+        tempTTStub.addClusterRef( edmNew::makeRefTo( clusterHandle, lowerClusterIter ) );
+        tempTTStub.addClusterRef( edmNew::makeRefTo( clusterHandle, upperClusterIter ) );
+
+        /// Check for compatibility
+        bool thisConfirmation = false;
+        int thisDisplacement = 999999;
+        int thisOffset = 0;
+
+        theStubFindingAlgoHandle->PatternHitCorrelation( thisConfirmation, thisDisplacement, thisOffset, tempTTStub );
+
+        /// If the Stub is above threshold
+        if ( thisConfirmation )
+        {
+          tempTTStub.setTriggerDisplacement( thisDisplacement );
+          tempTTStub.setTriggerOffset( thisOffset );
+	  tempOutput.push_back( tempTTStub );
+        } /// Stub accepted
+      } /// End of loop over outer clusters
+
+      /// Here tempOutput stores all the stubs from this inner cluster
+      /// Check if there is need to store only one (if only one already, skip this step)
+      if ( ForbidMultipleStubs && tempOutput.size() > 1 )
+      {
+        /// If so, sort the stubs by bend and keep only the first one (smallest bend)
+        std::sort( tempOutput.begin(), tempOutput.end(), TTStubBuilder< Ref_Phase2TrackerDigi_ >::SortStubsBend );
+
+        /// Get to the second element (the switch above ensures there are min 2)
+        typename std::vector< TTStub< Ref_Phase2TrackerDigi_ > >::iterator tempIter = tempOutput.begin();
+        ++tempIter;
+
+        /// tempIter points now to the second element
+
+        /// Delete all-but-the first one from tempOutput
+        tempOutput.erase( tempIter, tempOutput.end() );
+      }
+
+      /// Here, tempOutput is either of size 1 (if entering the switch)
+      /// either of size N with all the valid combinations ...
+
+      /// Now loop over the accepted stubs (1 or N) for this inner cluster
+      for ( unsigned int iTempStub = 0; iTempStub < tempOutput.size(); ++iTempStub )
+      {
+        /// Get the stub
+        TTStub< Ref_Phase2TrackerDigi_ > tempTTStub = tempOutput.at( iTempStub );
+
+        /// Put in the output
+        if ( maxStubs == 0 )
+        {
+          /// This means that ALL stubs go into the output
+          tempInner.push_back( *(tempTTStub.getClusterRef(0)) );
+          tempOuter.push_back( *(tempTTStub.getClusterRef(1)) );
+          tempAccepted.push_back( tempTTStub );
+        }
+        else
+        {
+          /// This means that only some of them do
+          /// Put in the temporary output
+          int chip = tempTTStub.getTriggerPosition() / chipSize; /// Find out which ASIC
+          if ( moduleStubs.find( chip ) == moduleStubs.end() ) /// Already a stub for this ASIC?
+          {
+            /// No, so new entry
+            std::vector< TTStub< Ref_Phase2TrackerDigi_ > > tempStubs;
+            tempStubs.clear();
+            tempStubs.push_back( tempTTStub );
```

Comment:
these three lines can just be
std::vector< TTStub< Ref_Phase2TrackerDigi_ > > tempStubs(1,tempTTStub);